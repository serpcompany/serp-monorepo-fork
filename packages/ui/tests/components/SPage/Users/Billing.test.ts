import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it, vi } from 'vitest';
import { ref } from 'vue';
import Billing from '../../../../components/SPage/Users/Billing.vue';
import ComponentRender from '../../../componentRender';

// Declare a mutable session_ variable.
let session_: { loggedIn: unknown; user: unknown; clear: unknown };

mockNuxtImport('useUserSession', () => () => session_);
mockNuxtImport('useSeoMeta', () => () => {});

describe('SPage/Users/Billing Snapshot', () => {
  const scenarios: [
    string,
    { session: { loggedIn: unknown; user: unknown; clear: unknown } }
  ][] = [
    [
      'when user is logged in',
      {
        session: {
          loggedIn: ref(true),
          user: ref({
            name: 'Test User',
            email: 'test@test.com',
            image: 'https://example.com/test.jpg'
          }),
          clear: vi.fn()
        }
      }
    ],
    [
      'when user is not logged in',
      {
        session: {
          loggedIn: ref(false),
          user: ref(null),
          clear: vi.fn()
        }
      }
    ]
  ];

  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, { session }) => {
      // Update the session for this scenario.
      session_ = session;

      const html = await ComponentRender(
        `SPage/Users/Billing ${desc}`,
        {},
        Billing
      );
      expect(html).toMatchSnapshot();
    }
  );
});
