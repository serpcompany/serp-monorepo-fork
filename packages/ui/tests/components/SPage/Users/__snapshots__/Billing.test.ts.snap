// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPage/Users/Billing Snapshot > renders when user is logged in correctly 1`] = `
"<div class="flex flex-col lg:grid lg:grid-cols-10 lg:gap-10">
  <!--v-if-->
  <div class="lg:col-span-10">
    <div class="w-full px-4 sm:px-6 lg:px-8 mx-auto mt-48 max-w-3xl">
      <div class="rounded-lg bg-default ring ring-default divide-y divide-default my-8">
        <div class="p-4 sm:px-6">
          <div class="flex items-center justify-between">
            <h2 class="text-xl font-semibold">Billing</h2><span class="font-medium inline-flex items-center text-xs px-2 py-1 gap-1 rounded-md bg-primary/10 text-primary"><!--v-if-->Active<!--v-if--></span>
          </div>
        </div>
        <div class="p-4 sm:p-6">
          <div class="px-4 text-center">
            <p class="mb-6 text-gray-500 dark:text-gray-400"> Manage your subscription, view purchase history, or update payment methods. </p><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary mx-auto"><span class="iconify i-lucide:credit-card shrink-0 size-5" aria-hidden="true"></span><span class="truncate">Manage Billing</span>
              <!--v-if-->
            </button>
          </div>
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`SPage/Users/Billing Snapshot > renders when user is not logged in correctly 1`] = `
"<div class="flex flex-col lg:grid lg:grid-cols-10 lg:gap-10">
  <!--v-if-->
  <div class="lg:col-span-10">
    <div class="w-full px-4 sm:px-6 lg:px-8 mx-auto mt-48 max-w-3xl">
      <div class="rounded-lg bg-default ring ring-default divide-y divide-default my-8">
        <div class="p-4 sm:px-6">
          <div class="flex items-center justify-between">
            <h2 class="text-xl font-semibold">Billing</h2><span class="font-medium inline-flex items-center text-xs px-2 py-1 gap-1 rounded-md bg-primary/10 text-primary"><!--v-if-->Active<!--v-if--></span>
          </div>
        </div>
        <div class="p-4 sm:p-6">
          <div class="px-4 text-center">
            <p class="mb-6 text-gray-500 dark:text-gray-400"> Manage your subscription, view purchase history, or update payment methods. </p><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 text-inverted bg-primary hover:bg-primary/75 disabled:bg-primary aria-disabled:bg-primary focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary mx-auto"><span class="iconify i-lucide:credit-card shrink-0 size-5" aria-hidden="true"></span><span class="truncate">Manage Billing</span>
              <!--v-if-->
            </button>
          </div>
        </div>
        <!--v-if-->
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
